src/core/address_finder.rs

```diff
@@ -152,19 +147,18 @@ mod os_impl {
     use std;
     use read_process_memory::*;
 
-    pub fn current_thread_address(
+    pub fn current_vm_address(
         pid: pid_t,
-        version: &str,
-        is_maybe_thread: Box<Fn(usize, usize, ProcessHandle, &Vec<MapRange>) -> bool>,
+        is_maybe_vm: Box<Fn(usize, ProcessHandle) -> bool>,
     ) -> Result<usize, Error> {
         let proginfo = &get_program_info(pid)?;
-        match current_thread_address_symbol_table(proginfo, version) {
+        match current_vm_address_symbol_table(proginfo) {
             Some(addr) => Ok(addr),
             None => {
                 debug!("Trying to find address location another way");
```

Comment:
do you log anywhere that you were trying to find it in the symbol table? if not "another way" is kind of vague